alex-alexhollender/homework_10/1_callback_exercise/main.js

```diff
@@ -0,0 +1,96 @@
+// This file contains functions that use Ajax to request and log data from two different web services. Start by reading over the code and making sure you understand how it works.
+// Obtain API keys for both openweathermap.org and timezonedb.com. Specify the values the appropriate API keys in the weatherData and timeData objects, then verify that the Ajax requests complete successfully.
+// Next, update the app so the Ajax functionality is in a separate function that uses a callback to run other functions on the data it receives.
+// Finally, update your code so the Ajax function itself returns a function, then use it to create a getWeather() and a getTime() function.  
+
+'use strict';
+
+var weatherData = {};
+weatherData.apikey = "c78fdc82343747fbafa21a15424b715b";
+weatherData.url = "http://api.openweathermap.org/data/2.5/weather?q=San Francisco,CA&appid=" + weatherData.apikey;
+weatherData.category = "weather";
+
+var timeData = {};
+timeData.apikey = "0FHQE6S3VH8T";
+timeData.url = "http://api.timezonedb.com/v2/get-time-zone?key=" + timeData.apikey + "&format=json&by=zone&zone=America/Los_Angeles";
+timeData.category = "time";
+
+function updateWeather() {
+  $.ajax({
+      url: weatherData.url,
+      // Work with the response
+      success: function (r) {
+        var degF = (r.main.temp - 273.15) * 1.8 + 32;
+        var degFInt = Math.floor(degF);
+        console.log("Temperature: " + degFInt + " deg F");
+      },
+      error: function () {
+        console.log("There was an error getting weather data :(");
+      }
+  });
+}
+
+function updateTime() {
+  $.ajax({
+      url: timeData.url,
+      // Work with the response
+      success: function (r) {
+        // Create a new JavaScript Date object based on the timestamp
+        // multiplied by 1000 so that the argument is in milliseconds, not seconds.
+        var date = new Date((r.timestamp - r.gmtOffset)*1000);
+        // Hours part from the timestamp
```

Comment:
Stellar commenting! It really helps when other developers are reading your code.